### PR TITLE
Add stack pack for Wix

### PR DESCRIPTION
**Pull request by alonkochba**

### CODE REVIEW

File: packs/wix.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  /** Additional description of a Lighthouse audit that tells the user to defer loading of non-critical third-party libraries, in the context of the Wix CMS platform. */
+  "render-blocking-resources": "Wix ensures that our resources are not render-blocking. When [using third-party code](https://support.wix.com/en/article/site-performance-using-third-party-code-on-your-site) on your site, it’s important to add it in a non-blocking manner. For example, you can add defer or async attributes to your third-party script. You can also place the script at the end of the code body to ensure that your regular site content doesn’t get delayed or prevented from rendering.",
```
Comment (adamraine) [inline packs/wix.js:9]: This advice is not specific to wix sites. If there isn't some wix specific mechanism to defer scripts then we should remove this string.

File: packs/wix.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  /** Additional description of a Lighthouse audit that tells the user how they can improve performance by avoiding animated images and using videos where possible, in the context of the Wix CMS platform. */
+  "efficient-animated-content": "Using GIFs can significantly slow down a page's loading speed. Where possible, use videos instead of GIFs to showcase animated content. Modern video encoding formats are both higher quality and smaller in size than GIF animations. [Learn more](https://support.wix.com/en/article/site-performance-optimizing-your-media#optimizing-videos-on-your-site) about optimizing your site's media.",
```
Comment (adamraine) [inline packs/wix.js:11]: This advice is not specific to wix sites. If there isn't some wix specific mechanism here then we should remove this string.

File: packs/wix.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  /** Additional description of a Lighthouse audit that tells the user to be aware of JavaScript code that is not used, particularly from third-parties, in the context of the Wix CMS platform. */
+  "unused-javascript": "Third-party code is one of the main reasons for unused JavaScript. Identify these files and review whether they're necessary to your site. If not, consider removing or replacing them with an optimized alternative. Any code delivered through static.parastorage.com is packaged by Wix. It's crucial to your site's structure, so it can't be modified or deleted. We are consistently optimizing these files using methods such as code splitting, dynamic imports and using web workers to reduce main thread blocking JavaScript in order to minimize input delays and make sure that sites load quickly, regardless of their complexity. [Learn more](https://support.wix.com/en/article/site-performance-optimizing-your-media#optimizing-videos-on-your-site).",
```
Comment (adamraine) [inline packs/wix.js:13]: What tools does wix provide for users that are failing this audit? Telling users to remove unused services is pretty general advice that Lighthouse already suggests. The part about code splitting/dynamic imports doesn't identify any action the user can take.

Also I think the link is incorrect here.

File: packs/wix.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  /** Additional description of a Lighthouse audit that tells the user how they can improve the time to first byte speed metric, in the context of the Wix CMS platform. */
+  "server-response-time': 'Wix utilizes CDNs and caching to serve responses as fast as possible for most visitors. Try re-running the test. If response time is still high, consider [manually enabling caching](https://support.wix.com/en/article/site-performance-caching-pages-to-optimize-loading-speed) for your site, especially if using Velo.',
```
Comment (adamraine) [inline packs/wix.js:15]:
```suggestion
  "server-response-time': 'Wix utilizes CDNs and caching to serve responses as fast as possible for most visitors. Consider [enabling caching](https://support.wix.com/en/article/site-performance-caching-pages-to-optimize-loading-speed) for your site.',
```

File: packs/wix.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  /** Additional description of a Lighthouse audit that tells the user why 'block' font-display is used by default, in the context of the Wix CMS platform. */
+  'font-display': 'Wix uses the block font-display instead of showing a fallback font by default, which prevents visual instability—making for a smoother user experience. In any case, if the chosen font hasn’t loaded in a short period of time, most browsers will use a fallback font.',
```
Comment (adamraine) [inline packs/wix.js:17]: What user action can be taken on wix if this audit fails?

**Review by adamraine — COMMENTED** [review]
I dug through some of the docs linked here, and there is good advice. However most of these strings either don't provide any actionable advice for the user, or provide some general advice that Lighthouse already covers.

**Comment by alonkochba** [comment]
@adamraine I updated the proposed text, could you please have an additional look?

File: packs/wix.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  /** Additional description of a Lighthouse audit that tells the user how they can improve performance by avoiding animated images and using videos where possible, in the context of the Wix CMS platform. */
+  "efficient-animated-content": "Use videos instead of GIFs to showcase animated content for higher quality and faster load time. Place videos inside VideoBoxes, customize them using Video Masks or add Transparent Videos. [Learn more](https://support.wix.com/en/article/wix-video-about-wix-video).",
+  /** Additional description of a Lighthouse audit that tells the user to be aware of JavaScript code that is not used, particularly from third-parties, in the context of the Wix CMS platform. */
+  "unused-javascript": "Third-party code is one of the main reasons for unused JavaScript. Review any third-party code you've added to your site in the Custom Code tab of your site's dashboard and only keep the services that are necessary to your site. [Find out more](https://support.wix.com/en/article/site-performance-removing-unused-javascript).",
```
Comment (adamraine) [inline packs/wix.js:11]:
```suggestion
  "unused-javascript": "Review any third-party code you've added to your site in the `Custom Code` tab of your site's dashboard and only keep the services that are necessary to your site. [Find out more](https://support.wix.com/en/article/site-performance-removing-unused-javascript).",
```

File: packs/wix.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  /** Additional description of a Lighthouse audit that tells the user to defer loading of non-critical third-party libraries, in the context of the Wix CMS platform. */
+  "render-blocking-resources": "When [adding third-party code](https://support.wix.com/en/article/site-performance-using-third-party-code-on-your-site) in the Custom Code tab of your site's dashboard, make sure it's deferred or loaded at the end of the code body. Where possible, use Wix’s [integrations](https://support.wix.com/en/article/about-marketing-integrations) to embed marketing tools on your site. ",
+  /** Additional description of a Lighthouse audit that tells the user how they can improve performance by avoiding animated images and using videos where possible, in the context of the Wix CMS platform. */
+  "efficient-animated-content": "Use videos instead of GIFs to showcase animated content for higher quality and faster load time. Place videos inside VideoBoxes, customize them using Video Masks or add Transparent Videos. [Learn more](https://support.wix.com/en/article/wix-video-about-wix-video).",
```
Comment (adamraine) [inline packs/wix.js:9]: Surrounding product names with backticks
```suggestion
  "efficient-animated-content": "Place videos inside `VideoBoxes`, customize them using `Video Masks` or add `Transparent Videos`. [Learn more](https://support.wix.com/en/article/wix-video-about-wix-video).",
```

File: packs/wix.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+const UIStrings = {
+/** Additional description of a Lighthouse audit that tells the user to optimize image formats, in the context of the Wix CMS platform. */
```
Comment (adamraine) [inline packs/wix.js:4]:
```suggestion
  /** Additional description of a Lighthouse audit that tells the user to optimize image formats, in the context of the Wix CMS platform. */
```

File: packs/wix.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  "efficient-animated-content": "Use videos instead of GIFs to showcase animated content for higher quality and faster load time. Place videos inside VideoBoxes, customize them using Video Masks or add Transparent Videos. [Learn more](https://support.wix.com/en/article/wix-video-about-wix-video).",
+  /** Additional description of a Lighthouse audit that tells the user to be aware of JavaScript code that is not used, particularly from third-parties, in the context of the Wix CMS platform. */
+  "unused-javascript": "Third-party code is one of the main reasons for unused JavaScript. Review any third-party code you've added to your site in the Custom Code tab of your site's dashboard and only keep the services that are necessary to your site. [Find out more](https://support.wix.com/en/article/site-performance-removing-unused-javascript).",
+  /** Additional description of a Lighthouse audit that tells the user how they can improve the time to first byte speed metric, in the context of the Wix CMS platform. */
```
Comment (adamraine) [inline packs/wix.js:12]:
```suggestion
  /** Additional description of a Lighthouse audit that tells the user how they can improve the server response time, in the context of the Wix CMS platform. */
```

File: packs/wix.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+const UIStrings = {
+/** Additional description of a Lighthouse audit that tells the user to optimize image formats, in the context of the Wix CMS platform. */
+  "modern-image-formats": "Upload images using Wix Media Manager to ensure they are automatically served as WebP. Find [more ways to optimize](https://support.wix.com/en/article/site-performance-optimizing-your-media) your site's media.",
```
Comment (adamraine) [inline packs/wix.js:5]: These strings go through our translation pipeline, however the links you have provided will always be the english version of the wix articles. We shouldn't translate these product titles if that's the case so we surround them with backticks to signal that to translators.

```suggestion
  "modern-image-formats": "Upload images using `Wix Media Manager` to ensure they are automatically served as WebP. Find [more ways to optimize](https://support.wix.com/en/article/site-performance-optimizing-your-media) your site's media.",
```

File: packs/wix.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+  /** Additional description of a Lighthouse audit that tells the user to defer loading of non-critical third-party libraries, in the context of the Wix CMS platform. */
+  "render-blocking-resources": "When [adding third-party code](https://support.wix.com/en/article/site-performance-using-third-party-code-on-your-site) in the Custom Code tab of your site's dashboard, make sure it's deferred or loaded at the end of the code body. Where possible, use Wix’s [integrations](https://support.wix.com/en/article/about-marketing-integrations) to embed marketing tools on your site. ",
```
Comment (adamraine) [inline packs/wix.js:7]:
```suggestion
  "render-blocking-resources": "When [adding third-party code](https://support.wix.com/en/article/site-performance-using-third-party-code-on-your-site) in the `Custom Code` tab of your site's dashboard, make sure it's deferred or loaded at the end of the code body. Where possible, use Wix’s [integrations](https://support.wix.com/en/article/about-marketing-integrations) to embed marketing tools on your site. ",
```

File: packs/wix.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+  /** Additional description of a Lighthouse audit that tells the user to be aware of JavaScript code that is not used, particularly from third-parties, in the context of the Wix CMS platform. */
+  "unused-javascript": "Third-party code is one of the main reasons for unused JavaScript. Review any third-party code you've added to your site in the Custom Code tab of your site's dashboard and only keep the services that are necessary to your site. [Find out more](https://support.wix.com/en/article/site-performance-removing-unused-javascript).",
+  /** Additional description of a Lighthouse audit that tells the user how they can improve the time to first byte speed metric, in the context of the Wix CMS platform. */
+  "server-response-time": "Wix utilizes CDNs and caching to serve responses as fast as possible for most visitors. Consider [manually enabling caching](https://support.wix.com/en/article/site-performance-caching-pages-to-optimize-loading-speed) for your site, especially if using Velo.",
```
Comment (adamraine) [inline packs/wix.js:13]:
```suggestion
  "server-response-time": "Wix utilizes CDNs and caching to serve responses as fast as possible for most visitors. Consider [manually enabling caching](https://support.wix.com/en/article/site-performance-caching-pages-to-optimize-loading-speed) for your site, especially if using `Velo`.",
```

**Comment by alonkochba** [comment]
> These strings go through our translation pipeline, however the links you have provided will always be the english version of the wix articles. We shouldn't translate these product titles if that's the case so we surround them with backticks to signal that to translators.

Thanks, good to know.

We do have translations of some of these articles, would it be possible to provide a list of links for available languages?

For example https://support.wix.com/en/article/site-performance-optimizing-your-media is also available in 11 other languages: https://support.wix.com/es/article/rendimiento-del-sitio-optimizar-los-archivos-multimedia

(Press the small globe on the header to see languages)

**Comment by adamraine** [comment]
> We do have translations of some of these articles, would it be possible to provide a list of links for available languages?

We cannot provide multiple versions of links in these strings. The wix docs would need to automatically detect the user locale and serve the correct language. We would likely also want to have 100% overlap between supported locales.

**Comment by adamraine** [comment]
@alonkochba tests are failing, can you add a wix item to https://github.com/GoogleChrome/lighthouse-stack-packs/blob/master/index.js

**Comment by alonkochba** [comment]
> @alonkochba tests are failing, can you add a wix item to https://github.com/GoogleChrome/lighthouse-stack-packs/blob/master/index.js

Sure, added, thanks!

**Comment by adamraine** [comment]
> Is there any way to see some kind of deploy preview of this?

I don't think so unfortunately.

**Pull request merged**